syntax = "proto3";
option optimize_for = LITE_RUNTIME;

message Monitor {
  message AudioChannel {
    int32 audioClippingCount = 1;
    float levelFast = 2;
    float levelSlow = 3;
  }

  message AudioStats {
    repeated AudioChannel audioChannels = 1;
    int32 streamBufferPos = 2;
    int32 bufferOverrunCount = 3;
    int32 bufferUnderrunCount = 4;
    int32 codecErrorCount = 5;
  }

  message MuxChannelStats {
    int32 dupBlockCount = 1;
    int32 oooBlockCount = 2;
    int32 lastBlockSbnDiff = 3;
    int32 dupPacketCount = 4;
    int32 oooPacketCount = 5;

    oneof specificStats {
      AudioStats audioStats = 6;
    }
  }

  repeated MuxChannelStats muxChannels = 1;
}
